### PR TITLE
Fix discovered hosts tests

### PR DESCRIPTION
**Pull request by svtkachenko**
Changes:
* Medium was added to OS and hostgroup, otherwise autoprovision was failing with `Error: Failed to auto provision host macde7d1b5acb4d: Medium can't be blank`
* Common `select_action_dropdown` locator was updated to match nested elements with entity name, e.g. `//td/span/a` and not only `//td/span`.
* For `test_positive_update_name` `name = gen_string('alpha').lower()` is reduced to lower case as Satellite displays hosts in lowercase and search was failing.

Test results:
```
py.test -v --junit-xml=foreman-results.xml -m 'not stubbed' tests/foreman/ui/test_discoveredhost.py
============================= test session starts ==============================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/jenkins/shiningpanda/jobs/ad0ef6b5/virtualenvs/d41d8cd9/bin/python2.7
cachedir: .cache
rootdir: /home/jenkins/workspace/satellite6-standalone-automation, inifile:
plugins: xdist-1.20.0, services-1.2.1, mock-1.6.2, forked-0.2
collecting ... collected 58 items
2017-08-31 02:08:01 - conftest - DEBUG - Found WONTFIX in decorated tests ['1147100', '1269196', '1378009', '1245334', '1217635', '1226425', '1156555', '1311113', '1199150', '1204686', '1221971', '1103157', '1230902', '1230865', '1214312', '1079482']


tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_provision_with_hostgroup_from_new_model_window <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_pxe_based_discovery <- robottelo/decorators/__init__.py FAILED
tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_update_name <- robottelo/decorators/__init__.py FAILED

2017-08-31 02:08:01 - conftest - DEBUG - Collected 58 test cases
============================= 35 tests deselected ==============================
============ 3 failed, 20 passed, 35 deselected in 6381.26 seconds =============
```

Re-run of failed tests:
```
% pytest --html=report.html -m 'not stubbed' tests/foreman/ui/test_discoveredhost.py -k 'test_positive_pxe_based_discovery or test_positive_provision_with_hostgroup_from_new_model_window or test_positive_update_name'
=============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0
rootdir: /home/qui/code/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.0, html-1.12.0, cov-2.5.1
collected 58 items 
2017-08-31 11:23:09 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_discoveredhost.py  ...

-------------------------------------------- generated html file: /home/qui/code/robottelo/report.html --------------------------------------------
=============================================================== 55 tests deselected ===============================================================
=================================================== 3 passed, 55 deselected in 1004.61 seconds ====================================================
```
Test results for other tests that use `common_locators["select_action_dropdown"]`
```
% pytest -v --html=report.html tests/foreman/ui/test_{computeprofile,computeresource,discoveryrule,environment,host,hostgroup,location,organization,role}.py -k 'test_positive_delete and not positive_delete_'
d=============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.0, html-1.12.0, cov-2.5.1
collected 200 items 
2017-08-31 11:55:24 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_computeprofile.py::ComputeProfileTestCase::test_positive_delete PASSED
tests/foreman/ui/test_computeresource.py::ComputeResourceTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveryrule.py::DiscoveryRuleTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_environment.py::EnvironmentTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_host.py::HostTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_hostgroup.py::HostgroupTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_location.py::LocationTestCase::test_positive_delete <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_organization.py::OrganizationTestCase::test_positive_delete PASSED
tests/foreman/ui/test_role.py::RoleTestCase::test_positive_delete PASSED

-------------------------------------------- generated html file: /home/qui/code/robottelo/report.html --------------------------------------------
============================================================== 191 tests deselected ===============================================================
=================================================== 9 passed, 191 deselected in 1241.05 seconds ===================================================
```

### CODE REVIEW

File: robottelo/api/utils.py [hunk 1]
```diff
@@ -406,15 +406,20 @@ def configure_provisioning(org=None, loc=None, compute=False):
         query={u'search': u'name="x86_64"'}
     )[0].read()
 
+    #
```
Comment (oshtaier) [inline robottelo/api/utils.py]: ?

File: robottelo/api/utils.py [hunk 2]
```diff
@@ -406,15 +406,20 @@ def configure_provisioning(org=None, loc=None, compute=False):
...
+    media = entities.Media(location=[loc], organization=[org]).create()
```
Comment (oshtaier) [inline robottelo/api/utils.py]: real media should be provided automatically, not the fake one

Comment (svtkachenko) [reply robottelo/api/utils.py]: @oshtaier @sghai Ok, this was fixed for 6.2 but never was backported to master, I mean for 6.2 there is added media while code in master has nothing related to media. I'll backport that a little bit later.

**Comment by svtkachenko** [comment]
@ldjebran @oshtaier I've 'forwardport'ed  changes from 6.2.z, now there is no fake media. Please re-review.

Test results.
```
% pytest -v -s --html=report.html -m 'not stubbed' tests/foreman/ui/test_discoveredhost.py -k 'test_positive_auto_provision_all or test_positive_delete or test_positive_delete_from_facts or test_positive_delete_multiple or test_positive_manual_provision_host_with_rule or test_positive_provision_from_factstest_positive_provision_using_quick_host_button or test_positive_provision_with_hostgroup_from_new_model_window or test_positive_reboot or test_positive_refresh_facts_pxe or test_positive_refresh_facts_pxe_less or test_positive_update_default_location or test_positive_update_default_org or test_positive_update_name'

=============================================================== 46 tests deselected ===============================================================
============================================== 4 failed, 8 passed, 46 deselected in 3083.09 seconds ===============================================
```
Failed tests:
```
test_positive_refresh_facts_pxe_less
test_positive_refresh_facts_pxe
test_positive_delete_from_facts
test_positive_auto_provision_all
```
2 of them failed because of infrastructure issue.
Re-run for 2 others:
```
% pytest -v --html=report.html -m 'not stubbed' tests/foreman/ui/test_discoveredhost.py -k 'test_positive_auto_provision_all or test_positive_delete_from_facts'
=============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.13, pytest-3.0.7, py-1.4.34, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.0, html-1.12.0, cov-2.5.1
collected 58 items 
2017-09-11 16:57:33 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_auto_provision_all <- robottelo/decorators/__init__.py PASSED
tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_delete_from_facts <- robottelo/decorators/__init__.py PASSED

-------------------------------------------- generated html file: /home/qui/code/robottelo/report.html --------------------------------------------
=============================================================== 56 tests deselected ===============================================================
==================================================== 2 passed, 56 deselected in 911.95 seconds ====================================================
```

**Comment by svtkachenko** [comment]
@oshtaier I've updated PR, now `kickstart_repository` is used instead of media.

Result for test that was failing with 'medium can't be blank' error.
```
% pytest -v --html=report.html -m 'not stubbed' tests/foreman/ui/test_discoveredhost.py -k 'test_positive_auto_provision_all'
=============================================================== test session starts ===============================================================
platform linux2 -- Python 2.7.13, pytest-3.1.3, py-1.4.34, pluggy-0.4.0 -- /home/qui/code/venv/2/bin/python2
cachedir: .cache
rootdir: /home/qui/code/robottelo, inifile:
plugins: xdist-1.18.1, services-1.2.1, mock-1.6.2, html-1.12.0, cov-2.3.1
collected 58 items 
2017-09-15 15:52:01 - conftest - DEBUG - Deselect of WONTFIX BZs is disabled in settings


tests/foreman/ui/test_discoveredhost.py::DiscoveryTestCase::test_positive_auto_provision_all <- robottelo/decorators/__init__.py PASSED

-------------------------------------------- generated html file: /home/qui/code/robottelo/report.html --------------------------------------------
=============================================================== 57 tests deselected ===============================================================
==================================================== 1 passed, 57 deselected in 770.88 seconds ====================================================
```

File: robottelo/api/utils.py [hunk 3]
```diff
@@ -465,7 +458,7 @@ def configure_provisioning(org=None, loc=None, compute=False):
         puppet_proxy=proxy,
         puppet_ca_proxy=proxy,
         content_source=proxy,
-        medium=media,
+        kickstart_repository=repo.id,
```
Comment (sghai) [inline robottelo/api/utils.py]: 👍

**Comment by sghai** [comment]
ACK.. looks great. thank you @svtkachenko

**Pull request merged**